### PR TITLE
Adding Azure Object Anchors Conversion error code field

### PR DESCRIPTION
**Pull request by jastenze**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Generated/Models/ConversionErrorCode.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using System.ComponentModel;
+
+namespace Azure.MixedReality.ObjectAnchors.Conversion.Models
+{
+    /// <summary> The ConversionErrorCode. </summary>
+    public readonly partial struct ConversionErrorCode : IEquatable<ConversionErrorCode>
+    {
+        private readonly string _value;
+
+        /// <summary> Determines if two <see cref="ConversionErrorCode"/> values are the same. </summary>
```
Comment (JoshLove-msft) [inline sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Generated/Models/ConversionErrorCode.cs:18]: This description seems misplaced. /cc @ShivangiReja

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.MixedReality.ObjectAnchors.Conversion.Models
+{
+    /// <summary> The ConversionErrorCode. </summary>
+    public readonly partial struct ConversionErrorCode
```
Comment (JoshLove-msft) [inline sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:10]: Why was this needed as opposed to relying on the code gen?

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        [CodeGenMember("NOError")]
+        public static ConversionErrorCode NoError { get; } = new ConversionErrorCode(NoErrorValue);
+        /// <summary> SERVICE_ERROR. </summary>
+        [CodeGenMember("ServiceError")]
+        public static ConversionErrorCode ServiceError { get; } = new ConversionErrorCode(ServiceErrorValue);
+        /// <summary> INVALID_ASSET_URI. </summary>
+        [CodeGenMember("InvalidAssetURI")]
+        public static ConversionErrorCode InvalidAssetUri { get; } = new ConversionErrorCode(InvalidAssetUriValue);
+        /// <summary> INVALID_JOB_ID. </summary>
+        [CodeGenMember("InvalidJOBID")]
```
Comment (JoshLove-msft) [inline sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:25]: Why is this cased like this?

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        public static ConversionErrorCode InvalidJobId { get; } = new ConversionErrorCode(InvalidJobIdValue);
+        /// <summary> INVALID_GRAVITY. </summary>
+        [CodeGenMember("InvalidGravity")]
+        public static ConversionErrorCode InvalidGravity { get; } = new ConversionErrorCode(InvalidGravityValue);
+        /// <summary> INVALID_SCALE. </summary>
+        [CodeGenMember("InvalidScale")]
+        public static ConversionErrorCode InvalidScale { get; } = new ConversionErrorCode(InvalidScaleValue);
+        /// <summary> ASSET_SIZE_TOO_LARGE. </summary>
+        [CodeGenMember("AssetSizeTOOLarge")]
```
Comment (JoshLove-msft) [inline sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: Similar question about the casing.

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.MixedReality.ObjectAnchors.Conversion.Models
+{
+    /// <summary> The ConversionErrorCode. </summary>
```
Comment (JoshLove-msft) [inline sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:9]: If we need this custom model we should update the docs.

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 3 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:25]: So autogen for some reason turned INVALID_JOB_ID into InvalidJOBID when generating the extensible enum, I guess it thought anything with <4 letters in that uppercase underscore-delimited error code string was an acronym? The custom partial class was for renaming them with proper casing

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 4 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: So autogen for some reason turned ASSET_SIZE_TOO_LARGEinto AssetSizeTOOLarge when generating the extensible enum, I guess it thought anything with <4 letters in that uppercase underscore-delimited error code string was an acronym? The custom partial class was for renaming them with proper casing

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 2 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:10]: Casing needed to be fixed for a lot of these

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 4 above)
Comment (JoshLove-msft) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: /cc @ShivangiReja @pakrym is this a known issue?

Comment (pakrym) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: cc @chamons

Comment (pakrym) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: I know we have special logic in namer to process acronyms. @timotheeguerin does this sound familiar?

Comment (timotheeguerin) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: yeah modelerfour naming, will keep upper words together up to a certain length. there is way to configure around if this is not what you want. In this case I think Just chancing the x-ms-client-name to be `AssetSizeTooLarge` or `Asset_Size_Too_Large` should have it done correctly.

Comment (JoshLove-msft) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs:34]: @jastenze can we use swagger transforms to set the x-ms-client-name instead of the current approach?

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 2 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs]: Fixed through swagger change

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 3 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs]: Fixed through swagger change

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 4 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs]: Fixed through swagger change

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs (hunk 5 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Customizations/Models/ConversionErrorCode.cs]: Removed through swagger change, but docs do need updating

**Review by JoshLove-msft — APPROVED** [review]
I assume the plan is to eventually update the swagger in the REST repo and that this is a temporary workaround?
LGTM

File: sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Generated/Models/ConversionErrorCode.cs (hunk 1 above)
Comment (jastenze) [reply sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/Generated/Models/ConversionErrorCode.cs:18]: Agreed, but requires autogen fix
